### PR TITLE
Add support for AWS Cognito

### PR DESCRIPTION
**Pull request by simonrondelez**
## Description

This PR adds support for AWS Cognito to AWS elasticsearch clusters

## Related Issue

https://github.com/skyscrapers/engineering/issues/237

## Motivation and Context

needed to allow users to use the Cognito authentication on Kibana

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -33,6 +33,26 @@ variable "options_indices_query_bool_max_clause_count" {
   default     = "1024"
 }
 
+variable "cognito_enabled" {
+  description = "Bool(optional, false): Whether to enable Cognito for authentication in Kibana"
+  default     = false
+}
+
+variable "cognito_user_pool_id" {
+  description = "String(required) ID of the Cognito User Pool to use"
```
Comment (venturel) [inline variables.tf]: Only a small thing, I would clarify that it's required if `cognito_enabled` is `true`, same for the other 2 vars

**Review by venturel — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ Terraform module to setup all resources needed for setting up an AWS Elasticsear
 * [`options_rest_action_multi_allow_explicit_index`]: String(optional, "true"): Sets the `rest.action.multi.allow_explicit_index` advanced option (must be string, not bool!). If you want to configure access to domain sub-resources, such as specific indices, you must set this property to "false". Setting this property to "false" prevents users from bypassing access control for sub-resources
```
Comment (duboisph) [inline README.md]: Maybe update the docs using `terraform-docs` :)

**Review by duboisph — APPROVED** [review]

File: variables.tf (hunk 1 above)
Comment (simonrondelez) [reply variables.tf]: indeed was doubting. Changed now

File: README.md (hunk 2 above)
Comment (simonrondelez) [reply README.md]: true :(

**Pull request merged**